pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -267,6 +266,7 @@ def on_init_end(self, trainer: "pl.Trainer") -> None:
     def on_pretrain_routine_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
         """When pretrain routine starts we build the ckpt dir on the fly."""
         self.__resolve_ckpt_dir(trainer)
+        self.__check_if_dir_not_empty(self.dirpath, trainer)
```

Comment:
```suggestion
        if trainer.is_global_zero:
            self.__warn_if_dir_not_empty(self.dirpath)
```
nit to keep the warning function more self-contained